Doc/howto/clinic.rst

```diff
@@ -1891,3 +1891,82 @@ blocks embedded in Python files look slightly different.  They look like this:
     #[python start generated code]*/
     def foo(): pass
     #/*[python checksum:...]*/
+
+
+How to deprecate positional use of optional parameters
+------------------------------------------------------
+
+Argument Clinic provides syntax that makes it possible to generate code that
+deprecates positional use of optional parameters.
+For example, say we've got a module level function :py:func:`!foo.myfunc` that
+takes three :class:`int` parameters:
+optional parameters *a* and *b*, and keyword-only parameter *c*::
+
+   /*[clinic input]
+   module foo
+   myfunc
+       a: int
+       b: int
+       *
+       c: int
+   [clinic start generated output]*/
+
+We now want to make the *b* parameter keyword-only from Python 3.14 and onward,
+and since :py:func:`!myfunc` is a public API,
+we must follow :pep:`387` and issue a deprecation warning.
+We can do that using the ``* [from ...]``` syntax,
+by adding the line ``* [from 3.14]`` right above the *b* parameter::
```

Comment:
It's a little unclear here whether Python 3.14 is the next version of Python to be released, one version in the future, two versions in the future...

I suggest being more explicit about what version of Python is currently represented by the CPython `main` branch in this example -- Python 3.14 will one day be the CPython `main` branch, but I don't want us to have to update this example every year :)